### PR TITLE
LibCrypto: Some clean-up of the SECP256r1 curve implementation

### PR DESCRIPTION
**Pull request by msvisser**
This PR contains some clean-up of the SECP256r1 curve implementation that I did a while ago. These clean-ups are mostly to improve readability and to make the code more generic in preparation for the SECP384r1 implementation.

Most significantly, the large list of `u256` constants at the top of the file is now reduced to 4 constants, which are taken directly from the curve specification. All the other constants are now calculated instead.

### CODE REVIEW

**Review by ADKaster — APPROVED** [review]
This looks good to me, but I'll let @alimpfard have a chance to look before merging

**Review by alimpfard — APPROVED** [review]

**Pull request merged**
